lib/googleauth/credentials.rb

```diff
@@ -373,8 +377,8 @@ def initialize keyfile, options = {}
         else
           update_from_filepath keyfile, options
         end
+        setup_logging logger: options.fetch(:logger, :default)
         @project_id ||= CredentialsLoader.load_gcloud_project_id
-        @client.fetch_access_token! if @client.needs_access_token?
```

Comment:
was this change intentional?

Reply:
@tartakynov Yes it was. Auth clients should be fetching tokens just-in-time when a request is actually made, rather than eagerly up front. Did this cause an issue?

Reply:
@dazuma yeah, I left a bug report here 👉  https://github.com/googleapis/google-auth-library-ruby/issues/508

Reply:
@dazuma This change also broke our codebase... We now get this error when trying to call BigQuery:
```rb
Google::Cloud::UnauthenticatedError: Unauthorized
```

Upgrading to `1.12.2` etc does not help.